db/versions/0037.yml

```diff
@@ -0,0 +1,333 @@
+      Update a queue provisioner.
+    mode: write
+    args: provisioner_id_in text, expires_in timestamptz, last_date_active_in timestamptz, description_in text, stability_in text, actions_in jsonb
+    returns: table(provisioner_id text, expires timestamptz, last_date_active timestamptz, description text, stability text, actions jsonb, etag uuid)
+    body: |-
+      declare
+        new_etag uuid := public.gen_random_uuid();
+      begin
+        return query update queue_provisioners
+        set
+          expires = expires_in,
+          last_date_active = last_date_active_in,
+          description = description_in,
+          stability = stability_in,
+          actions = actions_in,
+          etag = new_etag
+        where
+          queue_provisioners.provisioner_id = provisioner_id_in
+        returning
+          queue_provisioners.provisioner_id,
+          queue_provisioners.expires,
+          queue_provisioners.last_date_active,
+          queue_provisioners.description,
+          queue_provisioners.stability,
+          queue_provisioners.actions,
+          queue_provisioners.etag;
+      end
+  get_queue_provisioners:
+    description: |-
+      Get queue provisioners ordered by `provisioner_id`.
+      If the pagination arguments are both NULL, all rows are returned.
+      Otherwise, page_size rows are returned at offset page_offset.
+    mode: read
+    serviceName: queue
+    args: page_size_in integer, page_offset_in integer
+    returns: table(provisioner_id text, expires timestamptz, last_date_active timestamptz, description text, stability text, actions jsonb, etag uuid)
+    body: |-
+      begin
+        return query
+        select
+          queue_provisioners.provisioner_id,
+          queue_provisioners.expires,
+          queue_provisioners.last_date_active,
+          queue_provisioners.description,
+          queue_provisioners.stability,
+          queue_provisioners.actions,
+          queue_provisioners.etag
+        from queue_provisioners
+        order by provisioner_id
+        limit get_page_limit(page_size_in)
+        offset get_page_offset(page_offset_in);
+      end
+  expire_queue_provisioners:
+    description: |-
+      Expire provisioners that come before `expires_in`.
+      Returns a count of rows that have been deleted.
+    mode: write
+    serviceName: queue
+    args: expires_in timestamptz
+    returns: integer
+    body: |-
+      declare
+        count integer;
+      begin
+        delete from queue_provisioners
+        where queue_provisioners.expires < expires_in;
+        if found then
+          get diagnostics count = row_count;
+          return count;
+        end if;
+        return 0;
+      end
+  remove_queue_provisioners_table:
+    description: |-
+      Removes the queue provisioners table in case of backouts.
+    mode: write
+    serviceName: queue
+    args: []
+    returns: void
+    body: |-
+      begin
+        drop table queue_provisioners;
```

Comment:
Hm, I wonder if dropping the table would create issues. Truncating the table would probably make more sense here as well as for the other `remove_` functions defined in this PR.

Reply:
I think we can just get rid of all this `remove_.._table` stuff.